algorithms/src/std_algorithms/Kokkos_PartitioningOperations.hpp

```diff
@@ -0,0 +1,493 @@
+    OutputIteratorFalseType to_first_false, PredicateType p) {
+  return Impl::partition_copy_impl(label, ex, from_first, from_last,
+                                   to_first_true, to_first_false, std::move(p));
+}
+
+template <class ExecutionSpace, class DataType1, class... Properties1,
+          class DataType2, class... Properties2, class DataType3,
+          class... Properties3, class PredicateType>
+auto partition_copy(
+    const ExecutionSpace& ex,
+    const ::Kokkos::View<DataType1, Properties1...>& view_from,
+    const ::Kokkos::View<DataType2, Properties2...>& view_dest_true,
+    const ::Kokkos::View<DataType3, Properties3...>& view_dest_false,
+    PredicateType p) {
+  return Impl::partition_copy_impl("Kokkos::partition_copy_view_api_default",
+                                   ex, cbegin(view_from), cend(view_from),
+                                   begin(view_dest_true),
+                                   begin(view_dest_false), std::move(p));
+}
+
+template <class ExecutionSpace, class DataType1, class... Properties1,
+          class DataType2, class... Properties2, class DataType3,
+          class... Properties3, class PredicateType>
+auto partition_copy(
+    const std::string& label, const ExecutionSpace& ex,
+    const ::Kokkos::View<DataType1, Properties1...>& view_from,
+    const ::Kokkos::View<DataType2, Properties2...>& view_dest_true,
+    const ::Kokkos::View<DataType3, Properties3...>& view_dest_false,
+    PredicateType p) {
+  return Impl::partition_copy_impl(label, ex, cbegin(view_from),
+                                   cend(view_from), begin(view_dest_true),
+                                   begin(view_dest_false), std::move(p));
+}
+
+// ----------------------
+// partition_point
+// ----------------------
+template <class ExecutionSpace, class IteratorType, class UnaryPredicate>
+IteratorType partition_point(const ExecutionSpace& ex, IteratorType first,
+                             IteratorType last, UnaryPredicate p) {
+  return Impl::partition_point_impl(
+      "Kokkos::partitioned_point_iterator_api_default", ex, first, last,
+      std::move(p));
+}
+
+template <class ExecutionSpace, class IteratorType, class UnaryPredicate>
+IteratorType partition_point(const std::string& label, const ExecutionSpace& ex,
+                             IteratorType first, IteratorType last,
+                             UnaryPredicate p) {
+  return Impl::partition_point_impl(label, ex, first, last, std::move(p));
+}
+
+template <class ExecutionSpace, class UnaryPredicate, class DataType,
+          class... Properties>
+auto partition_point(const std::string& label, const ExecutionSpace& ex,
+                     const ::Kokkos::View<DataType, Properties...>& v,
+                     UnaryPredicate p) {
+  Impl::static_assert_is_admissible_to_kokkos_std_algorithms(v);
+  return Impl::partition_point_impl(label, ex, cbegin(v), cend(v),
+                                    std::move(p));
```

Comment:
https://github.com/kokkos/kokkos/pull/4315#discussion_r711010047 still applies here and below.